lib/array_intersection.rb

```diff
@@ -1,4 +1,24 @@
+#Design and implement a method that takes two integer arrays with unique values and returns their intersection in a new array.
+
+#intersection([2, 3, 4], [4, 5, 6]) => [4]
+#intersection([50, 43, 25, 72], [25, 36, 43, 50, 80]) => [50, 25, 43]
+#intersection([9, 30, 42], [56, 34, 90, 32]) => []
+
 def intersection(list1, list2)
-  raise NotImplementedError, "Intersection not implemented"
+
+  listone_hash = {}
+  answer = []
+  
+  list1.each do |num|
+    listone_hash.has_value?(num) 
```

Comment:
You need to put elements in the hash before you check to see if they have the value.

This code doesn't make sense, and it's not working.

```suggestion
    listone_hash[num] = true
```